tests/phpunit/test-class-wc-stripe-upe-payment-method.php

```diff
@@ -285,6 +285,11 @@ public function test_card_payment_method_capability_is_always_enabled() {
 		$this->set_mock_payment_method_return_value( 'is_subscription_item_in_cart', false );
 		$this->set_mock_payment_method_return_value( 'get_capabilities_response', self::MOCK_INACTIVE_CAPABILITIES_RESPONSE );
 
+		// Disable testmode.
+		$stripe_settings             = get_option( 'woocommerce_stripe_settings' );
+		$stripe_settings['testmode'] = 'no';
+		update_option( 'woocommerce_stripe_settings', $stripe_settings );
```

Comment:
I think this is needed as a consequence of migrating from `setUp()/tearDown()` to `set_up()/tear_down()`.
 
Maybe @lovo-h has more info on this (https://github.com/woocommerce/woocommerce-gateway-stripe/commit/fe650c1a8baac40bb1a0429c1e5268c51c6e341f)

Reply:
Although I am not 100% certain, I believe that the reason this test worked **before** is that `testmode` was already set to `no` due to some other factor. Or, it was unset and was empty and therefore, it defaulted to `no` (as I demonstrate below).

Fast forward to **now** and that "other factor" is making it so that `testmode` is already set to `yes`. To summarize: before, the test was working by coincidence. So, it's better to be explicit about setting the value of `testmode` in order to avoid situations like this where the result is inconsistent.

Now, the reason I believe it's some "other factor" is that if one were to run the test by itself, it passes just fine without needing to be explicit about the value of `testmode`. But, if we run the test suite (e.g. `npm run test:php`), then the test fails. So, something is setting the `testmode` value before this test has a chance to run.

---

In order to better explain what was going on, we can take a brief look at the code. For a non-`card` payment method, if we follow the stack trace, we get something like the following:
```
class-wc-stripe-upe-payment-method.php:211, WC_Stripe_UPE_Payment_Method->is_capability_active()
class-wc-stripe-upe-payment-method.php:155, WC_Stripe_UPE_Payment_Method->is_enabled_at_checkout()
test-class-wc-stripe-upe-payment-method.php:305, WC_Stripe_UPE_Payment_Method_Test->test_card_payment_method_capability_is_always_enabled()
```

And if we look at the following method in the stack trace `class-wc-stripe-upe-payment-method.php:211, WC_Stripe_UPE_Payment_Method->is_capability_active()`:
```php
public function is_capability_active() {
    ...
    // [ Note: if 'testmode' is empty, it defaults to 'no' (as mentioned above). ]
    $test_mode_setting = ! empty( $plugin_settings['testmode'] ) ? $plugin_settings['testmode'] : 'no';

    // [ Note: If we are not explicit about the value of 'testmode', some other factor might set it to 'yes'. ]
    if ( 'yes' === $test_mode_setting ) {
        return true;
    }

    // [ Note: 'get_capabilities_response()' is mocked for this test. ]
    $capabilities = $this->get_capabilities_response();
    ...
}
```

So, if `testmode` is already set to `yes`, the `is_capability_active()` method would return `true` and never reach the [mocked `get_capabilities_response()` method call](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2302/files#diff-358342362ae0a24ddc1254fd5f6632bff220c4147eaafffa3d5f0b7654ce3c11L286).

Finally, if the `is_capability_active()` method returns `true`, it ends up making `is_enabled_at_checkout()` return `true`, and ultimately we arrive at the incorrect conclusion: `$this->assertFalse( true )`.

That was a bit long-winded but I hope that makes sense 🙂

Reply:
Thanks for the explanation @lovo-h